src/imitation/zero_one_loss.jl

```diff
@@ -0,0 +1,58 @@
+"""
+    zero_one_loss(y, y_true)
+
+0-1 loss for multiclass classification: `δ(y, y_true) = 0` if `y = y_true`, and `1` otherwise.
+"""
+function zero_one_loss(y::AbstractArray, y_true::AbstractArray)
+    return y == y_true ? zero(eltype(y)) : one(eltype(y))
+end
+
+"""
+    zero_one_loss_maximizer(y, y_true; α)
+
+For `δ = zero_one_loss`, compute
+```
+argmax_y {δ(y, y_true) + α θᵀ(y - y_true)}
+```
+"""
+function zero_one_loss_maximizer(
+    θ::AbstractVector,
+    y_true::AbstractVector{R},  # TODO: does it work with arrays?
+    α,
+) where {R<:Real}
+    i_true = findfirst(==(one(R)), y_true)
+    i_θ = argmax(θ)
+    y = zeros(R, size(y_true))
+    if (i_true == i_θ) || (α * θ[i_true] > 1 + α * θ[i_θ])
+        y[i_true] = one(R)
+    else
+        y[i_θ] = one(R)
+    end
+    return y
+end
+
+"""
+    ZeroOneStructuredSVMLoss
+
+Implementation of the `StructuredSVMLoss` based on a 0-1 loss for multiclass classification.
+"""
+function ZeroOneStructuredSVMLoss(α=1)
+    return StructuredSVMLoss(;
+        aux_loss_maximizer=zero_one_loss_maximizer, δ=zero_one_loss, α=α
+    )
+end
+
+"""
+    ZeroOneStructuredSVMLoss(α)
+
+Implementation of the `ImitationLoss` based on a 0-1 loss for multiclass classification with no regularization.
+"""
+function ZeroOneImitationLoss(α=1)
```

Comment:
Is this equivalent to the above `ZeroOneStructuredSVMLoss` ? In this case, we don't really need it

Reply:
Actually I was wondering whether we could make both SPO and SSVM rely on our generic imitation loss. It would be theoretically pleasant

Reply:
I don't know if it's possible without losing some performance. The FenchelYoungLoss is also a special case of imitation loss

Reply:
Let's keep it the way it is for now

Reply:
And yes it is equivalent but it uses ImitationLoss instead, and allows us to test it

Reply:
We could therefore only define it in the tests